### PR TITLE
Adds the Power Fist

### PR DESCRIPTION
**Pull request by MMMiracles**
:cl: MMMiracles
add: Added the power fist, a gauntlet with a piston-powered ram attached to the top for traitors to punch people across the station. Punch people into walls for added fun.
/:cl:
# [Video example of in-game use can be seen here.](https://youtu.be/tpp9dY4uCp0)

wowlookatthosesprites
WOW LOOK AT THOSE SPRITES EVEN THOUGH THERE'S ALREADY A VIDEO

Merge token(s): #18998

The power-fist is a semi-port of hippiestation's power-fist. Sprites are from hippiestation.

The power fist is a metallic gauntlet with a built-in piston-ram that uses an external tank as its gas source. A wrench can be used to dial up the piston's valve and make the gauntlet hit harder and farther in exchange for more gas used.

A full large-tank will get exactly 4 level-3 punches in before needing to replace the external tank. This thing is incredibly lethal for taking out singular targets at a time but very poor against crowds as odds are you're going to exhaust your supply before fending off everyone.

The TC price might be too low/high, I don't know yet until people play around with it in an actual round.  All I know is punching pun-pun half-way down the escape hall in a single hit is fun.

### CODE REVIEW

**Comment by kevinz000** [comment]
ROOOCKEEET PUUUNCH

**Comment by evsey9** [comment]
Can you fist using the power fist?
Also, please make it electricity/magic powered, don't want to recharge it every 4 uses

**Comment by MMMiracles** [comment]
@evsey9 
If you recharge it every 4 uses that means you're using the max setting, which can 3-shot people.

Exchanging extra gas for raw power. Carry more tanks if you want to beat up people.

@optimumtact IT LITERALLY HAS A VIDEO THAT'S A BUNCH OF PICTURES PUT TOGETHER

**Comment by TechnoAlchemisto** [comment]
Doesn't show the item sprites

**Comment by MMMiracles** [comment]
..but it does?

reeeeee

**Comment by optimumtact** [comment]
Ayyy

**Comment by optimumtact** [comment]
LMAOOOOOOOOOOOOOOOO

**Comment by optimumtact** [comment]
alien hot air balloon rises in distant, faint ayy lmao can be heard

File: code/game/objects/items/weapons/powerfist.dm [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+		if(tank)
+			updateTank(tank, 1, user)
+
+
+/obj/item/weapon/melee/powerfist/proc/updateTank(obj/item/weapon/tank/internals/thetank, removing = 0, mob/living/carbon/human/user)
+	if(removing)
+		if(!src.tank)
+			return
+		user << "<span class='notice'>You detach \the [thetank] from \the [src].</span>"
+		src.tank.loc = get_turf(user)
```
Comment (optimumtact) [inline code/game/objects/items/weapons/powerfist.dm]: forceMove

File: code/game/objects/items/weapons/powerfist.dm [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+		src.tank = null
+	if(!removing)
+		if(src.tank)
+			user << "<span class='warning'>\The [src] already has a tank.</span>"
+			return
+		if(!user.unEquip(thetank))
+			return
+		user << "<span class='notice'>You hook \the [thetank] up to \the [src].</span>"
+		src.tank = thetank
+		thetank.loc = src
```
Comment (optimumtact) [inline code/game/objects/items/weapons/powerfist.dm]: forceMove

File: code/game/objects/items/weapons/powerfist.dm [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+	if(!in_range(user, src))
+		user << "<span class='notice'>You'll need to get closer to see any more.</span>"
+		return
+	if(tank)
+		user << "<span class='notice'>\icon [tank] It has \the [tank] mounted onto it.</span>"
+
+
+/obj/item/weapon/melee/powerfist/attackby(obj/item/weapon/W, mob/user, params)
+	if(istype(W, /obj/item/weapon/tank/internals))
+		if(!tank)
```
Comment (optimumtact) [inline code/game/objects/items/weapons/powerfist.dm]: maybe give some feedback that it already has a tank?

File: code/game/objects/items/weapons/powerfist.dm [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+				fisto_setting = 1
+		user << "<span class='notice'>You tweak \the [src]'s piston valve to [fisto_setting].</span>"
+	else if(istype(W, /obj/item/weapon/screwdriver))
+		if(tank)
+			updateTank(tank, 1, user)
+
+
+/obj/item/weapon/melee/powerfist/proc/updateTank(obj/item/weapon/tank/internals/thetank, removing = 0, mob/living/carbon/human/user)
+	if(removing)
+		if(!src.tank)
```
Comment (optimumtact) [inline code/game/objects/items/weapons/powerfist.dm]: no need for the src

**Comment by Super3222** [comment]
I feel like on output level 3 it should do some damage back onto the user, seeing as the pneumatic cannon does the same thing.

Considering it's an 8 TC weapon that has the potential to kill you in two strikes a drawback would be nice, but then again you could just raise the TC cost if you wanna keep it drawback free.

**Comment by MMMiracles** [comment]
It is a high-damage weapon, much like the already-existing energy sword. The pricing I have is based on the fact that while it can do more damage than the energy sword, it comes with the massive drawback of having to carry multiple tanks if you want to go around punching multiple people half-way across the station.

The balance between the cost is "Do you want a more consistent weapon that does less damage or a higher-damage weapon that requires extra maintenance to be effective?" I believe its fine as is but I won't be for certain until in-game use is seen.

**Comment by KorPhaeron** [comment]
may as well try it out

**Comment by TechnoAlchemisto** [comment]
MERGE

**Comment by Cheridan** [comment]
Esword itself is 2 hit crit so it's basically the same

**Comment by MMMiracles** [comment]
Yeah but punching people half-way across the station and that sick extra 6 damage.

**Pull request merged**